distribution/pull_v2_windows.go

```diff
@@ -55,3 +61,55 @@ func (ld *v2LayerDescriptor) open(ctx context.Context) (distribution.ReadSeekClo
 	}
 	return rsc, err
 }
+
+func filterManifests(manifests []manifestlist.ManifestDescriptor) []manifestlist.ManifestDescriptor {
+	version := system.GetOSVersion()
+
+	// TODO @jhowardmsft LCOW Support: Need to remove the hard coding in LCOW mode.
+	lookingForOS := runtime.GOOS
+	osVersion := fmt.Sprintf("%d.%d.%d", version.MajorVersion, version.MinorVersion, version.Build)
+	if system.LCOWSupported() {
+		lookingForOS = "linux"
+		osVersion = ""
+	}
+
+	var matches []manifestlist.ManifestDescriptor
+	for _, manifestDescriptor := range manifests {
+		if manifestDescriptor.Platform.Architecture == runtime.GOARCH && manifestDescriptor.Platform.OS == lookingForOS {
+			if !versionMatch(manifestDescriptor.Platform.OSVersion, osVersion) {
+				continue
+			}
+			matches = append(matches, manifestDescriptor)
+
+			logrus.Debugf("found match for %s/%s with media type %s, digest %s", runtime.GOOS, runtime.GOARCH, manifestDescriptor.MediaType, manifestDescriptor.Digest.String())
+		}
+	}
+	sort.Stable(manifestsByVersion(matches))
+	return matches
+}
+
+func versionMatch(actual, expected string) bool {
+	return actual == "" || expected == "" || actual == expected || strings.HasPrefix(expected, actual+".")
```

Comment:
Assuming the variable names are correct, they should be swapped in the prefix check:
```go
strings.HasPrefix(actual, expected+".")
```